code-studio/src/js/callouts.js

```diff
@@ -112,16 +124,49 @@ module.exports = function createCallouts(callouts) {
     }
 
     if (callout.on) {
-      $(window).on(callout.on, function () {
-        if (!callout.seen && $(selector).length > 0) {
+      $(window).on(callout.on, function (e, action) {
+        if (!config.codeStudio.selector) {
+          config.codeStudio.selector = selector;
+        }
+        var lastSelector = config.codeStudio.selector;
+        $(window).trigger('prepareforcallout', [config.codeStudio]);
+        if (lastSelector !== config.codeStudio.selector && $(lastSelector).length > 0) {
+          $(lastSelector).qtip(config).qtip('destroy');
+        }
+        if ($(config.codeStudio.selector).length > 0) {
+          if (action === 'hashchange' || action === 'hashinit' || !callout.seen) {
+            $(config.codeStudio.selector).qtip(config).qtip('show');
+          }
           callout.seen = true;
-          $(selector).qtip(config).qtip('show');
         }
       });
-    } else {
+    } else if (!callout.seen) {
       $(selector).qtip(config).qtip('show');
     }
   });
+
+  // Insert a hashchange handler to detect triggercallout= hashes and fire
+  // appropriate events to open the callout
+  function detectTriggerCalloutOnHash(event) {
```

Comment:
The `#triggercallout=` is intended to be used by markdown in the instructions to force callouts to appear. I attached an example video. Note how the droplet palette category changes automatically, the block scrolls into view, and the callouts now track the block as the palette scrolls.

The reason it doesn't use query parameters is so that it doesn't result in a trip back to the server - this is intended to be a client-side only URL change. But by using a standard URL format, it makes it easy to create links inside markdown.